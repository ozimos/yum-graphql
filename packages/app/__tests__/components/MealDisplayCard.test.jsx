/* global React:false, shallow:false toJson:false */

import MealsDisplayCard from '../../src/components/mealCard/MealDisplayCard';
import { meal } from '../__mocks__/mealDataMock';

const props = {
  ...meal,
  collection: 'menu'
};

describe('MealsDisplayCard Component', () => {

  const setup = () => shallow(<MealsDisplayCard {...props} />);

  it('renders correctly', () => {
    let wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();

    const propsWithAddButton = { ...props, addToCollection: jest.fn() };
    const setupWithAddButton = () => shallow(<MealsDisplayCard
      {...propsWithAddButton}
    />);
    wrapper = setupWithAddButton();
    expect(toJson(wrapper)).toMatchSnapshot();

    const propsWithRemoveButton = { ...props, removeFromCollection: jest.fn() };
    const setupWithRemoveButton = () => shallow(<MealsDisplayCard
      {...propsWithRemoveButton}
    />);
    wrapper = setupWithRemoveButton();
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper = setupWithAddButton();
    wrapper.find('button.btn.title-button').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper = setupWithRemoveButton();
    wrapper.find('button.btn.title-button').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

