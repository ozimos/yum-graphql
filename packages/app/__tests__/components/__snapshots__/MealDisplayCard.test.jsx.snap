// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealsDisplayCard Component renders correctly 1`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          >
            nice
          </span>
          <span>
            ₦
            200
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
        src="nice"
      />
    </div>
    <p
      className="descrip long_string"
    >
      nice
    </p>
  </div>
</React.Fragment>
`;

exports[`MealsDisplayCard Component renders correctly 2`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          >
            nice
          </span>
          <span>
            ₦
            200
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
        src="nice"
      />
    </div>
    <p
      className="descrip long_string"
    >
      nice
    </p>
    <button
      className="btn title-button"
      onClick={[Function]}
    >
      Add to menu
    </button>
  </div>
</React.Fragment>
`;

exports[`MealsDisplayCard Component renders correctly 3`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          >
            nice
          </span>
          <span>
            ₦
            200
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
        src="nice"
      />
    </div>
    <p
      className="descrip long_string"
    >
      nice
    </p>
    <button
      className="btn title-button"
      onClick={[Function]}
    >
      Remove from menu
    </button>
  </div>
</React.Fragment>
`;

exports[`MealsDisplayCard Component renders correctly 4`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          >
            nice
          </span>
          <span>
            ₦
            200
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
        src="nice"
      />
    </div>
    <p
      className="descrip long_string"
    >
      nice
    </p>
    <button
      className="btn title-button"
      onClick={[Function]}
    >
      Add to menu
    </button>
  </div>
</React.Fragment>
`;

exports[`MealsDisplayCard Component renders correctly 5`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          >
            nice
          </span>
          <span>
            ₦
            200
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
        src="nice"
      />
    </div>
    <p
      className="descrip long_string"
    >
      nice
    </p>
    <button
      className="btn title-button"
      onClick={[Function]}
    >
      Remove from menu
    </button>
  </div>
</React.Fragment>
`;
